backend/src/controllers/actorController.js

```diff
@@ -27,34 +29,58 @@ export const createActor = (req, res) => {
   newActor.save((err, actor) => {
     if (err) {
       if (err.name === 'ValidationError') {
-        res.status(422).send(err);
+        res.status(StatusCodes.UNPROCESSABLE_ENTITY).send(err);
       } else {
-        res.status(500).send(err);
+        res.status(StatusCodes.INTERNAL_SERVER_ERROR).send(err);
       }
     } else {
       res.json(actor);
     }
   });
 };
 
-export const updateActor = (req, res) => {
-  actorModel.findOneAndUpdate({ _id: req.params.actorId }, req.body, { new: true }, (err, actor) => {
-    if (err) {
-      if (err.name === 'ValidationError') {
-        res.status(422).send(err);
-      } else {
-        res.status(500).send(err);
-      }
+export const updateActor = async (req, res) => {
+  try {
+    
+    const { actor } = res.locals;
+
+    let access = false;
+
+    if (!actor) {
+      res.status(StatusCodes.UNAUTHORIZED).send('Not authorized');
+    }
+
+    if(actor.role === Roles.ADMIN) {
+       access = true;
+    } else if ( actor.id === req.params.actorId) {
+       access = true;
     } else {
-      res.json(actor);
+      res.status(StatusCodes.FORBIDDEN).send('The Actor is trying to update an Actor that is not himself!');
     }
-  });
+    
+    if (access) {
+      actorModel.findOneAndUpdate({ _id: req.params.actorId }, req.body, { new: true }, (err, actor) => {
+        if (err) {
+          if (err.name === 'ValidationError') {
+            res.status(StatusCodes.UNPROCESSABLE_ENTITY).send(err);
+          } else {
+            res.status(StatusCodes.INTERNAL_SERVER_ERROR).send(err);
+          }
+        } else {
+          res.json(actor);
+        }
+      });
```

Comment:
I'd refactor this to use async/await instead of callbacks to make it a bit more readable